### DISCUSSION TITLE
How to use mini app in TG bot

### DISCUSSION THREAD

**Discussion by rsgold222** [Schweinepriester/github-profile-achievements#931] [Q&A] [answered]
I'm using a mini app to display a list of products to a TG bot, but I can't get messages from the mini app.
I would appreciate some advice.

**Comment by venuswhispers** [comment, marked as answer]
TG bots can only receive messages from mini-apps when a user opens the app using the keyboard and menu.
It cannot receive messages when the app is opened with the inline keyboard.
https://topco.medium.com/from-bots-to-mini-apps-unveiling-telegrams-web3-ecosystem-edea254f9b5f

**Comment by rsgold222** [comment]
thanks very much.
